### PR TITLE
locust: Migrate to faster http client

### PR DESCRIPTION
**Pull request by akashin**
This yields around 30% CPU usage reduction when measured with 2 processes and 1000 users on FT workload.

### CODE REVIEW

**Review by bowenwang1996 — APPROVED** [review]

**Review by tayfunelmas — APPROVED** [review]
I also started using FastHttpUser and FastHttpSession but it is great that the entire requests API can be replaced with geteventhttpclient (which Fast* also uses). LGTM.

File: pytest/tests/loadtest/locust/common/base.py [hunk 1]
```diff
@@ -322,6 +322,43 @@ def send_tx(self, tx: Transaction, locust_name: str) -> dict:
         self.request_event.fire(**meta)
         return meta
 
+    def send_tx_async(self, tx: Transaction, locust_name: str) -> dict:
```
Comment (tayfunelmas) [inline pytest/tests/loadtest/locust/common/base.py:325]: is this change part of this PR? I thought you had another PR for using async API

Comment (akashin) [reply pytest/tests/loadtest/locust/common/base.py]: No, it appeared here because of the base branch that I've used. Should be fixed now after the rebase.

**Comment by akashin** [comment]
> I also started using FastHttpUser and FastHttpSession but it is great that the entire requests API can be replaced with geteventhttpclient (which Fast* also uses). LGTM.

I've tried `FastHttpUser` but had a problem with Locust dashboard as it started reporting requests twice, once for the events firing inside `FastHttpUser` and once for the events that we manually fire in `send_tx`. So I looked at the workaround for this.

**Pull request merged**